File: src/pages/rentalproperty.js
import React, { useState } from 'react';
import axios from 'axios';
import '../components/renatalmanagement.css';

const RentalProperty = () => {
  const [property_Type, setProperty_Type] = useState('');
  const [location, setLocation] = useState('');
  const [price_Range, setPrice_Range] = useState('');
  const [numberof_Bedrooms, setNumberof_Bedrooms] = useState('');
  const [property_size, setProperty_size] = useState('');
  const [floor_plan, setFloor_plan] = useState('');
  const [lease_Terms, setLease_Terms] = useState('');
  const [contact, setContact] = useState('');
  const [parking_Availabilty, setParking_Availabilty] = useState('');
  const [maintantence_fess,setMaintantence_fess] = useState('');
  const [availabilty_Status, setAvailabilty_Status] = useState('');
  const [ rent_Amount, setRent_Amount] = useState('');
  const [availabilty_Date, setAvailabilty_Date] = useState('');

  const handleSubmit = async (event) => {
    event.preventDefault();

    const rentalData = {
      property_Type,
      location,
      price_Range,
      numberof_Bedrooms,
      property_size,
      floor_plan,
      lease_Terms,
      contact,
      parking_Availabilty,
      maintantence_fess,
      availabilty_Status,
      rent_Amount,
      availabilty_Date,
    };

    try {
      await axios.post('http://localhost:8080/api/Rentalmanagement', rentalData);
      alert('Rental property submitted successfully');
    } catch (error) {
      console.error('Error submitting rental property:', error);
    }
  };

  return (
    <div className="connttainer">
  <h1>Rental Property</h1>
  <form onSubmit={handleSubmit}>
    {/* <input type="text" placeholder="Property Type" onChange={(e) => setProperty_Type(e.target.value)} /> */}
    <label htmlFor="Property Type"><h5>Property Type</h5></label>
    <select id="Property Type" onChange={(e) => setProperty_Type(e.target.value)}>
      <option value="Apartment">Apartment</option>
      <option value="House">House</option>
    </select>
    <input type="text" placeholder="Location" onChange={(e) => setLocation(e.target.value)} />
    {/* <input type="text" placeholder="Price Range" onChange={(e) => setPrice_Range(e.target.value)} /> */}
    <label htmlFor="Price Range"><h5>Price Range</h5></label>
    <select id="Price Range" onChange={(e) => setPrice_Range(e.target.value)}>
      <option value="1000 to 3000">1000 to 3000</option>
      <option value="1000 to 5000">1000 to 5000</option>
      <option value="5000 to 10000">5000 to 10000</option>
      <option value="10000 to 20000">10000 to 20000</option>
      <option value="20000 to 50000">20000 to 50000</option>
      <option value="50000 to 100000">50000 to 100000</option>
    </select>
    
    {/* <input type="text" placeholder="Number of Bedrooms" onChange={(e) => setNumberof_Bedrooms(e.target.value)} /> */}
   
    <label htmlFor="Number of Bedrooms"><h5>Number of Bedrooms</h5></label>
    <select id="Number of Bedrooms" onChange={(e) => setNumberof_Bedrooms(e.target.value)}>
      <option value="1">1</option>
      <option value="2">2</option>
      <option value="1">3</option>
      <option value="2">4</option>
      <option value="1">5</option>
     
    </select>
   
    <input type="text" placeholder="Property Size" onChange={(e) => setProperty_size(e.target.value)} />
    {/* <input type="text" placeholder="Floor Plan" onChange={(e) => setFloor_plan(e.target.value)} /> */}
    <input type="text" placeholder="Lease Terms" onChange={(e) => setLease_Terms(e.target.value)} />
    <input type="text" placeholder="Contact" onChange={(e) => setContact(e.target.value)} />

    <label htmlFor="Floor Plan"><h5>Floor Plan</h5></label>
    <select id="Floor Plan" onChange={(e) => setFloor_plan(e.target.value)}>
      <option value="1">1</option>
      <option value="2">2</option>
      <option value="1">3</option>
      <option value="2">4</option>
      <option value="1">5</option>
      <option value="2">6</option>
      <option value="1">7</option>
      <option value="2">8</option>
    </select>
    <label htmlFor="availabilityStatus"><h5>Availability Status</h5></label>
    <select id="availabilityStatus" onChange={(e) => setAvailabilty_Status(e.target.value)}>
      <option value="yes">Yes</option>
      <option value="no">No</option>
    </select>

    <label htmlFor="parkingAvailability"><h5>Parking Availability</h5></label>
    <select id="parkingAvailability" onChange={(e) => setParking_Availabilty(e.target.value)}>
      <option value="yes">Yes</option>
      <option value="no">No</option>
    </select>

    <label htmlFor="maintenanceFees"><h5>Maintenance Fees</h5></label>
    <select id="maintenanceFees" onChange={(e) => setMaintantence_fess(e.target.value)}>
      <option value="yes">Yes</option>
      <option value="no">No</option>
    </select>

    {/* <input type="text" placeholder="Rent Amount" onChange={(e) => setRent_Amount(e.target.value)} /> */}
    <h5>Availabilty Date</h5>
    <input type="date" placeholder="Availability Date" onChange={(e) => setAvailabilty_Date(e.target.value)} />

    <button type="submit">Submit</button>
  </form>
</div>


 
  );
};

export default RentalProperty;

File: Backend Code/src/main/java/com/example/demo/entity/RentalProperty.java
package com.example.demo.entity;

import java.sql.Date;

import org.springframework.format.annotation.DateTimeFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name="RentalProperty_Management")
public class RentalProperty {
	@Id
	@Column(name="Id")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long Id;
	private String Property_Type;
	private String Location;
	private String Price_Range;
	private String Numberof_Bedrooms;
	private String Property_size;
	private String Floor_plan;
	private String Lease_Terms;
	private String Contact;
	private String Parking_Availabilty;
	private String Maintantence_fess;
	private String Availabilty_Status;
	private String Rent_Amount;
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	private Date Availabilty_Date;
	
	
	public RentalProperty(long id, String property_Type, String location, String price_Range, String numberof_Bedrooms,
			String property_size, String floor_plan, String lease_Terms, String contact, String parking_Availabilty,
			String maintantence_fess, String availabilty_Status, String rent_Amount, Date availabilty_Date) {
		super();
		Id = id;
		Property_Type = property_Type;
		Location = location;
		Price_Range = price_Range;
		Numberof_Bedrooms = numberof_Bedrooms;
		Property_size = property_size;
		Floor_plan = floor_plan;
		Lease_Terms = lease_Terms;
		Contact = contact;
		Parking_Availabilty = parking_Availabilty;
		Maintantence_fess = maintantence_fess;
		Availabilty_Status = availabilty_Status;
		Rent_Amount = rent_Amount;
		Availabilty_Date = availabilty_Date;
	}


	public RentalProperty() {
		super();
		// TODO Auto-generated constructor stub
	}


	public long getId() {
		return Id;
	}


	public void setId(long id) {
		Id = id;
	}


	public String getProperty_Type() {
		return Property_Type;
	}


	public void setProperty_Type(String property_Type) {
		Property_Type = property_Type;
	}


	public String getLocation() {
		return Location;
	}


	public void setLocation(String location) {
		Location = location;
	}


	public String getPrice_Range() {
		return Price_Range;
	}


	public void setPrice_Range(String price_Range) {
		Price_Range = price_Range;
	}


	public String getNumberof_Bedrooms() {
		return Numberof_Bedrooms;
	}


	public void setNumberof_Bedrooms(String numberof_Bedrooms) {
		Numberof_Bedrooms = numberof_Bedrooms;
	}


	public String getProperty_size() {
		return Property_size;
	}


	public void setProperty_size(String property_size) {
		Property_size = property_size;
	}


	public String getFloor_plan() {
		return Floor_plan;
	}


	public void setFloor_plan(String floor_plan) {
		Floor_plan = floor_plan;
	}


	public String getLease_Terms() {
		return Lease_Terms;
	}


	public void setLease_Terms(String lease_Terms) {
		Lease_Terms = lease_Terms;
	}


	public String getContact() {
		return Contact;
	}


	public void setContact(String contact) {
		Contact = contact;
	}


	public String getParking_Availabilty() {
		return Parking_Availabilty;
	}


	public void setParking_Availabilty(String parking_Availabilty) {
		Parking_Availabilty = parking_Availabilty;
	}


	public String getMaintantence_fess() {
		return Maintantence_fess;
	}


	public void setMaintantence_fess(String maintantence_fess) {
		Maintantence_fess = maintantence_fess;
	}


	public String getAvailabilty_Status() {
		return Availabilty_Status;
	}


	public void setAvailabilty_Status(String availabilty_Status) {
		Availabilty_Status = availabilty_Status;
	}


	public String getRent_Amount() {
		return Rent_Amount;
	}


	public void setRent_Amount(String rent_Amount) {
		Rent_Amount = rent_Amount;
	}


	public Date getAvailabilty_Date() {
		return Availabilty_Date;
	}


	public void setAvailabilty_Date(Date availabilty_Date) {
		Availabilty_Date = availabilty_Date;
	}


	@Override
	public String toString() {
		return "RentalProperty [Id=" + Id + ", Property_Type=" + Property_Type + ", Location=" + Location
				+ ", Price_Range=" + Price_Range + ", Numberof_Bedrooms=" + Numberof_Bedrooms + ", Property_size="
				+ Property_size + ", Floor_plan=" + Floor_plan + ", Lease_Terms=" + Lease_Terms + ", Contact=" + Contact
				+ ", Parking_Availabilty=" + Parking_Availabilty + ", Maintantence_fess=" + Maintantence_fess
				+ ", Availabilty_Status=" + Availabilty_Status + ", Rent_Amount=" + Rent_Amount + ", Availabilty_Date="
				+ Availabilty_Date + "]";
	}
	
	
	
	

}
